src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,239 @@
+import {Injectable} from 'angular2/core';
+import {Http} from 'angular2/http';
+import {Observable} from 'rxjs/Rx';
+
+/**
+  *  Configuration for an icon, including the URL and possibly the cached SVG element.
+  */
+class IconConfig {
+  svgElement: SVGElement = null;
+  constructor(public url: string) {
+  }
+}
+
+@Injectable()
+export class MdIconRegistry {
+  // IconConfig objects and cached SVG elements for individual icons.
+  // First level of cache is namespace (which is the empty string if not specified).
+  // Second level is the icon name within the namespace.
+  private _iconConfigs = new Map<string, Map<string, IconConfig>>();
+
+  // IconConfig objects and cached SVG elements for icon sets.
+  // Multiple icon sets can be registered under the same namespace.
+  private _iconSetConfigs = new Map<string, IconConfig[]>();
+
+  // Cache for icons loaded by direct URLs.
+  private _cachedIconsByUrl = new Map<string, SVGElement>();
+
+  private _fontClassNamesByAlias = new Map<string, string>();
+
+  private _inProgressUrlFetches = new Map<string, Observable<string>>();
+
+  private _defaultFontSetClass = 'material-icons';
+
+  constructor(private _http: Http) {
+  }
+
+  public addIcon(iconName: string, url: string): this {
+    return this.addIconInNamespace('', iconName, url);
+  }
+
+  public addIconInNamespace(namespace: string, iconName: string, url: string): this {
+    let iconSetMap = this._iconConfigs.get(namespace);
+    if (!iconSetMap) {
+      iconSetMap = new Map<string, IconConfig>();
+      this._iconConfigs.set(namespace, iconSetMap);
+    }
+    iconSetMap.set(iconName, new IconConfig(url));
+    return this;
+  }
+
+  public addIconSet(url: string): this {
+      return this.addIconSetInNamespace('', url);
+  }
+
+  public addIconSetInNamespace(namespace: string, url: string): this {
+    const config = new IconConfig(url);
+    if (this._iconSetConfigs.has(namespace)) {
+      this._iconSetConfigs.get(namespace).push(config);
+    } else {
+      this._iconSetConfigs.set(namespace, [config]);
+    }
+    return this;
+  }
+
+  public registerFontSet(alias: string, className?: string): this {
```

Comment:
Actually if classNameForFontAlias returns its argument if it hasn't been registered as an alias, it seems like the second argument here doesn't need to be optional.